### PR TITLE
Test.functional

### PR DESCRIPTION
**Pull request by pjbreaux**
@zancas 

#### What's this change do?
Implements two suites of functional tests for F5's supported Heat templates. This does not necessarily mean we will promote those templates tested, as there is at least one known bug (https://github.com/F5Networks/f5-openstack-image-prep/issues/13). The two tested templates are the one for patching a ve image and the one for booting a ve with one management nic and two data nics. The only thing we are checking for when booting a ve is that it is licensed. This can be enhanced, but it is a starting point.

#### Any background context?
We need to begin promoting some of these templates to supported and this is the first step.

#### Where should the reviewer start?
Start at the functional tests and work around to the fixtures in each suite, then those in conftest.

### CODE REVIEW

**Comment by pjbreaux** [comment]
The pytest commands to run these tests are defined below. Fill in the details of your stack and the license key for a VE:

patch image tests:
     py.test test_patch_upload_ve_image.py  --auth-netloc= --os-username= --os-password= --os-tenant-id= --os-tenant-name= --no-teardown-glance-images -slvv

cli command for ve_boot:
     py.test test_f5_base_instance_deploy.py  --auth-netloc= --os-username= --os-password= --os-tenant-id= --os-tenant-name= --bigip-root-password= --bigip-admin-password= --license= -slv

File: test/functional/conftest.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+@pytest.fixture
+def OSPassword(request):
+    return request.config.getoption('--os-password')
+
+
+@pytest.fixture
+def AuthAddress(request):
+    return request.config.getoption('--auth-netloc')
```
Comment (zancas) [inline test/functional/conftest.py]: Hopefully these fixtures will migrate into f5-openstack-test.

File: test/functional/ve/standalone/test_f5_base_instance_deploy.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    bigip = BigIP(bigip_ip, username, password)
+    for attempt in range(max_attempts):
+        time.sleep(interval)
+        try:
+            registration = bigip.shared.licensing.registration.load()
+            assert registration.licensedVersion == bigip_version
+            return
+        except Exception:
+            continue
+
```
Comment (zancas) [inline test/functional/ve/standalone/test_f5_base_instance_deploy.py]: As you mention in your PR comments, eventually (as the test matures) we'll add more assertions.

**Comment by zancas** [comment]
NOTE:  The test lab went down before I could independently run your tests.   I'm going to merge this but I'd like you to sit next to me, and help me run the tests on my system (once the lab's back up).

File: test/functional/conftest.py (hunk 1 above)
Comment (pjbreaux) [inline test/functional/conftest.py]: Yeah I think they can. I think we'd probably want each config.getoption to be in separate fixtures. That way folks can compose them however they'd like.

**Comment by pjbreaux** [comment]
Thanks.

File: test/functional/conftest.py (hunk 1 above)
Comment (zancas) [inline test/functional/conftest.py]: +1 Would you please open an issue to track this?

**Pull request merged**